lib/src/blaze/block_witness_data.rs

```diff
@@ -372,6 +372,42 @@ impl BlockAndWitnessData {
             .remove_txns_at_height(reorg_height);
     }
 
+    pub async fn update_orchard_anchors_process(
+        &self,
+        start_block: u64,
+        end_block: u64,
+        uri: Arc<std::sync::RwLock<Uri>>,
+    ) -> JoinHandle<Result<(), String>> {
+        let orchard_anchors = self.orchard_anchors.clone();
+        tokio::spawn(async move {
+            let mut workers = FuturesUnordered::new();
+            for block_height in end_block..=start_block {
```

Comment:
Is it possible that this will end up consuming more threads of execution than is optimal?

Should we have some kind of cap on the number of workers, or is the interface to `workers.push` somehow managing this in a safe manner?

Reply:
I guess I see the `workers.push` inside loops in plenty of other places...